stl/inc/xmemory

```diff
@@ -2003,6 +2007,99 @@ typename _Container::size_type _Erase_nodes_if(_Container& _Cont, _Pr _Pred) {
     }
     return _Old_size - _Cont.size();
 }
+
+#if _HAS_CXX20
+template <class _Ty, class _Alloc, class... _Types, enable_if_t<!_Is_specialization_v<_Ty, pair>, int> = 0>
+constexpr auto uses_allocator_construction_args(const _Alloc& _Al, _Types&&... _Args) noexcept {
+    if constexpr (!uses_allocator_v<_Ty, _Alloc>) {
+        static_assert(is_constructible_v<_Ty, _Types...>,
+            "T must be constructible from Types... if uses_allocator_v<T, Alloc> is false");
+        (void) _Al;
+        return _STD forward_as_tuple(_STD forward<_Types>(_Args)...);
+    } else {
+        if constexpr (is_constructible_v<_Ty, allocator_arg_t, const _Alloc&, _Types...>) {
+            using _ReturnType = tuple<allocator_arg_t, const _Alloc&, _Types&&...>;
+            return _ReturnType(allocator_arg, _Al, _STD forward<_Types>(_Args)...);
+        } else if constexpr (is_constructible_v<_Ty, _Types..., const _Alloc&>) {
+            return _STD forward_as_tuple(_STD forward<_Types>(_Args)..., _Al);
+        } else {
+            static_assert(_Always_false<_Ty>,
+                "T must be constructible from either (allocator_arg_t, const Alloc&, Types...) "
+                "or (Types..., const Alloc&) if uses_allocator_v<T, Alloc> is true");

```

Comment:
Maybe simplify to `"T is not constructible from the provided types.`

Reply:
That's less informative, as it doesn't tell user what constructors are expected